File: common/src/main/java/juuxel/adorn/trading/TradeInventory.java
package juuxel.adorn.trading;

import juuxel.adorn.util.InventoryComponent;

public final class TradeInventory extends InventoryComponent {
    private final Trade trade;

    public TradeInventory(Trade trade) {
        super(2);
        this.trade = trade;

        setItem(0, trade.getSelling());
        setItem(1, trade.getPrice());

        addListener(() -> {
            trade.setSelling(getItem(0));
            trade.setPrice(getItem(1));
            trade.callListeners();
        });
    }

    public Trade getTrade() {
        return trade;
    }
}

File: src/main/kotlin/juuxel/adorn/trading/TradeInventory.kt
package juuxel.adorn.trading

import juuxel.adorn.util.InventoryComponent

class TradeInventory(val trade: Trade) : InventoryComponent(2) {
    init {
        setStack(0, trade.selling)
        setStack(1, trade.price)

        addListener {
            trade.selling = getStack(0)
            trade.price = getStack(1)
            trade.callListeners()
        }
    }
}
